api/src/main/java/ai/djl/util/ZipUtils.java

```diff
@@ -121,13 +120,31 @@ private static void addToZip(Path root, Path file, ZipOutputStream zos) throws I
         }
     }
 
+    static String sanitizeAndValidateArchiveEntry(String name, Path destination)
+            throws IOException {
+        String sanitizedName = removeLeadingFileSeparator(name);
+        Path expectedOutputPath = destination.resolve(sanitizedName).normalize();
+        if (!expectedOutputPath.startsWith(destination.normalize())) {
+            throw new IOException(
+                    "Bad archive entry "
+                            + name
+                            + ". Attempted write outside destination "
+                            + destination);
+        }
+        return sanitizedName;
+    }
+
     static String removeLeadingFileSeparator(String name) {
+        String osAwareArchiveEntryName = FilenameUtils.separatorsToSystem(name);
```

Comment:
1. try to avoid dependency on `commons.io`
2. If we always run `sanitizeAndValidateArchiveEntry()`, we only need to remove "/" char here, "\" issue will be caught by `sanitizeAndValidateArchiveEntry()`

Reply:
why avoid commons.io? we're using it in a few places in this code path already

Reply:
1. in `api` module, the only place use commons.io is in `TarUtils.java`, `commons-io` is transient dependency from `commons-compression` recently added in 1.27.x, which is not intention of api project
2. there are customers they don't use `tar` files, they can exclude `commons-compression` from their project. see: #2949